src/lib.rs

```diff
@@ -183,9 +160,79 @@ pub fn is_dev() -> Option<bool> {
 /// `Some(true)` if the running version supports features, and `Some(false)`
 /// otherwise.
 pub fn supports_features() -> Option<bool> {
-    match is_nightly() {
-        b@Some(true) => b,
-        _ => is_dev()
+    Rustc::detect().map(|r| r.supports_features())
+}
+
+/// A detected instance of `rustc`, internally containing version information.
+pub struct Rustc {
+    version: Option<String>,
+    date: Option<String>,
+}
+
+impl Rustc {
+    /// Execute `rustc --version`, detecting where rustc is installed as well as
+    /// its local version and date. The returned `Rustc`, if this succeeds, can
+    /// be used to query version information.
+    pub fn detect() -> Option<Rustc> {
+        env::var("RUSTC").ok()
+            .and_then(|rustc| Command::new(rustc).arg("--version").output().ok())
+            .or_else(|| Command::new("rustc").arg("--version").output().ok())
+            .and_then(|output| String::from_utf8(output.stdout).ok())
+            .map(|s| version_and_date_from_rustc_version(&s))
+            .map(|(version, date)| Rustc { version: version, date: date })
+    }
+
+    /// Same as the crate `is_min_date` function
```

Comment:
Can we have this (and the others that are similar) link to the docs for the free function?